### PR TITLE
Reference to '_1' is ambiguous

### PR DESCRIPTION
**Pull request by hfcrwx**
centos7-x86_64 编译器及库均为默认版本，有编译错误：
/root/muduo/contrib/thrift/ThriftConnection.cc:23:47: 错误：对‘_1’的引用有歧义
                                         this, _1, _2, _3));
原因：
using namespace muduo;引起muduo::_1和boost中(anonymous namespace)::_1的混乱
解决方法：
using namespace muduo; -> using muduo::Timestamp;

### CODE REVIEW

**Comment by chenshuo** [comment]
1. One patch per PR please.
2. Change `boost::bind` to `std::bind` is a better fix for `contrib/thrift/ThriftConnection.cc`

**Comment by hfcrwx** [comment]
contrib/thrift: boost::bind -> std::bind
centos7和ubuntu14.04中默认thrift版本依赖boost::shared_ptr，还没有更新到C++11版本

File: contrib/thrift/ThriftConnection.cc [hunk 1]
```diff
@@ -1,16 +1,15 @@
...
 #include "muduo/base/Logging.h"
 
 #include <thrift/transport/TTransportException.h>
 
 #include "contrib/thrift/ThriftServer.h"
 
-using muduo::Timestamp;
-using muduo::net::Buffer;
-using muduo::net::TcpConnectionPtr;
+using namespace muduo;
```
Comment (chenshuo) [inline contrib/thrift/ThriftConnection.cc]: Please don't `using namespace` in contrib code.

File: contrib/thrift/ThriftConnection.cc [hunk 2]
```diff
@@ -19,8 +18,8 @@ ThriftConnection::ThriftConnection(ThriftServer* server,
     state_(kExpectFrameSize),
     frameSize_(0)
 {
-  conn_->setMessageCallback(boost::bind(&ThriftConnection::onMessage,
-                                        this, _1, _2, _3));
+  conn_->setMessageCallback(std::bind(&ThriftConnection::onMessage,
+                                        this, muduo::_1, muduo::_2, muduo::_3));
```
Comment (chenshuo) [inline contrib/thrift/ThriftConnection.cc:22]: Replace `muduo::_1` with just `_1`, you may need to `using muduo::_1` at the beginning of this file.

Ditto for _2, _3 and below.

Comment (hfcrwx) [reply contrib/thrift/ThriftConnection.cc]: I have changed it as you suggest.

**Pull request merged**